Topics/Tech_Stacks/PostgreSQL_psycopg2.md

```diff
@@ -0,0 +1,74 @@
+# Learning PostgreSQL and psycopg2
+
+### Prerequisites
+Before learning about PostgreSQL and psycopg2, ensure that you have knowledge about the following:
+- [Python](https://www.python.org/): High level general purpose programming language.
+- Relational Algebra
+- Database Theory
+- Basic understanding of SQL
+
+### Introduction
+PostgreSQL is an open-source DBMS. As the name states, PostgreSQL is a relational DBMS based on relations and queries and is great for dealing with large amounts of data and complex queries. It's also good for referencing data quickly and allowing the user to have some flexibility in how they want to represent their data. Furthermore, PostgreSQL supports table inheritance and function overloading.
+
+pyscopg2 is an adapter for Python that allows you to easily integrate PostgreSQL into your program. It's commonly used with other Python libraries like Flask, which allows you to use a PostgreSQL database in your application easily.
+
+### PostgreSQL and psycopg2 installation
+The following link is to download PostgreSQL onto your computer: https://www.postgresql.org/download/ . This link also shows the different versions of PostgreSQL to match your computer. Follow the instructions after downloading. For your convenience, here are the links to download the installer for PostgreSQL on different OS:
+- Windows: https://www.postgresql.org/download/windows/
+- MacOS: https://www.postgresql.org/download/macosx/
+- Linux: https://www.postgresql.org/download/linux/
+
+Once installed, you can create a database, relation, and tables with the PostgreSQL client called "pgAdmin". You can also perform queries on relations and tables through its query tool. More information on how to use pgAdmin can be found from this [link](https://www.pgadmin.org/docs/pgadmin4/6.21/index.html).
```

Comment:
Some confusion about the “pgAdmin” client - does this come installed with the postgresql download? How do I access it (i.e. do I just search on my taskbar?). After these questions are answered, the link you provided on how to use it will suffice.

Reply:
Answered question in md file